frontend/src/metabase/dashboard/actions/data-fetching.js

```diff
@@ -276,7 +277,7 @@ export const fetchDashboard = createAsyncThunk(
       return {
         entities,
         dashboard: result,
-        dashboardId: dashId,
+        dashboardId: result.id,
```

Comment:
I've taken a look at all the dashboard types above.
- `embed`: Embedded dashboard
- `transient`: x-ray dashboard
- `inline`: I don't think this is used anymore. Asked [a question in Slack.](https://metaboat.slack.com/archives/C505ZNNH4/p1706106083493939)
- and a normal dashboard.

All have `result.id === dashId` except the embed dashboard since I change that to `IS_EMBED_PREVIEW ? result.id : dashId` in this PR. I don't want to repeat the same condition again, so I figure using `result.id` would make more sense.